storage/merkle/errors.go

```diff
@@ -0,0 +1,50 @@
+package merkle
+
+import (
+	"errors"
+	"fmt"
+)
+
+var ErrorIncompatibleKeyLength = errors.New("key has incompatible size")
+
+// MalformedProofError is returned when a proof format
+// has some issues (syntax checks).
+type MalformedProofError struct {
```

Comment:
would suggest to also include a method to test the error type:
```golang
// IsMalformedProofError checks if err is a IsMalformedProofError
func IsMalformedProofError(err error) bool {
	var target *MalformedProofError
	return errors.As(err, &target)
}
```

Reply:
added